src/aabb.rs

```diff
@@ -0,0 +1,175 @@
+//! Axis Aligned Bounding Boxes.
+
+use nalgebra::{Point3, Vector3};
+use std::f32;
+use std::ops::Index;
+
+/// AABB struct.
+#[derive(Debug, Copy, Clone)]
+pub struct AABB {
+    // Minimum coordinates
+    pub min: Point3<f32>,
+
+    // Maximum coordinates
+    pub max: Point3<f32>,
+}
+
+/// A trait implemented by things which can be bounded by an `AABB`.
+pub trait Bounded {
+    fn aabb(&self) -> AABB;
+}
+
+impl AABB {
+    /// Creates a new `AABB` with the given bounds.
+    pub fn new(min: Point3<f32>, max: Point3<f32>) -> AABB {
```

Comment:
`with_dimensions` is misleading. What you specify is not the position and dimension of the `AABB`, but the bounds. I will rename it to `with_bounds`.